src/main/java/com/uber/cadence/workflow/Workflow.java

```diff
@@ -21,11 +21,342 @@
 import com.uber.cadence.activity.ActivityOptions;
 import com.uber.cadence.common.RetryOptions;
 import com.uber.cadence.internal.sync.WorkflowInternal;
+import com.uber.cadence.workflow.Functions.Func;
+import com.uber.cadence.workflow.Functions.Func1;
+import com.uber.cadence.workflow.Functions.Func2;
+import com.uber.cadence.workflow.Functions.Proc;
 import java.time.Duration;
 import java.util.Optional;
+import java.util.UUID;
 import java.util.concurrent.CancellationException;
 import java.util.function.Supplier;
 
+/**
+ * Workflow encapsulates the orchestration of activities and child workflows. It can also answer to
```

Comment:
These comments are all in the readme, which I'm breaking down into docs. Is there a reason the information is repeated here? Let me know if I need to review this file also.

Reply:
These will end up in the generated JavaDoc. Java developers frequently avoid documentation and go directly to JavaDoc. I thought about removing them from the readme, but as JavaDoc is not very well organized having a separate well organized documentation makes sense. So I think we end up duplicating some of it.